pkg/eval/mods/file/file_test.go

```diff
@@ -30,7 +30,21 @@ func TestFile(t *testing.T) {
 				What:     "arguments here",
 				ValidLow: 1, ValidHigh: 1, Actual: 0}),
 
-		That(`p = (file:pipe); echo haha > $p; pwclose $p; slurp < $p; prclose $p`).Puts("haha\n"),
+		That("file:prclose").Throws(
+			errs.ArityMismatch{
+				What:     "arguments here",
+				ValidLow: 1, ValidHigh: 1, Actual: 0}),
+
+		That("file:pwclose").Throws(
+			errs.ArityMismatch{
+				What:     "arguments here",
+				ValidLow: 1, ValidHigh: 1, Actual: 0}),
+
+		That(`p = (file:pipe); echo haha > $p; pwclose $p
+		; slurp < $p; prclose $p`).Puts("haha\n"),
+
+		That(`p = (file:pipe); echo haha > $p; file:prclose $p
+		; head -n1 < $p`).Throws(AnyError),
```

Comment:
This fails because the external `head` command on FreeBSD doesn't exit with a non-zero status when the pipe is closed. In general it is best to avoid relying on external commands in unit tests .

Reply:
So what should I do differently? If I do not wish on removing this test case, what should I do differently?

Reply:
Replace the external `head` command with an appropriate builtin that will succeed if the read end of the pipe is open and fail, by raising an error, if it is closed. In an interactive elvish shell try a few builtins that read from stdin to find one that would be useful for this test.

Reply:
Thank you very much, will push the updated file today.